solr/core/src/java/org/apache/solr/client/solrj/embedded/EmbeddedSolrServer.java

```diff
@@ -251,7 +251,7 @@ public void writeResults(ResultContext ctx, JavaBinCodec codec) throws IOExcepti
                 .setWritableDocFields(resolver)
                 .marshal(rsp.getValues(), out);
 
-            try (InputStream in = out.toInputStream()) {
+            try (ByteArrayInputStream in = new ByteArrayInputStream(out.toByteArray())) {
```

Comment:
This is less efficient as it produces an extra byte array (and copies to it) than commons-io.  There's a hack around it though... see `org.apache.solr.client.solrj.impl.BinaryRequestWriter.BAOS` which shows that the original buffer can be accessible.  Maybe you could improve on that utility and use it here?

Reply:
Take a look at f03649c15ace75d8e3ce99960a9412dc386a7501 and see what you think

Reply:
Thanks @dsmiley I like the changes you made in c6cf531